test/test_automl/test_estimators.py

```diff
@@ -257,6 +258,42 @@ def test_type_of_target(mock_estimator):
                     "binary targets")
 
 
+def test_performance_over_time(tmp_dir, output_dir):
```

Comment:
This test looks great, but there is one thing I will like to suggest, which is try to re-use the existing test.

The point of my comment is that unit testing takes a big amount of time, 30+ mins, where the biggest contribution comes from fitting an AutoSklearnClassifier.

What if we instead of creating a new `test_performance_over_time` we add your checks `include_single_scores` and `includes_all_scores` into some existing functions. Maybe one suggestion will be:

* `includes_all_scores` --> [after this line](https://github.com/automl/auto-sklearn/blob/2e58f9df6eb39c3439bef40b390710b7504884c4/test/test_automl/test_estimators.py#L507)
* `include_single_scores` --> [after this line](https://github.com/automl/auto-sklearn/blob/2e58f9df6eb39c3439bef40b390710b7504884c4/test/test_automl/test_estimators.py#L546)

Also, maybe it makes sense to check if the dataframe is not empty and the embedded scores make sense (make sense means that the performance is increasing where we get a cumulative score or greater than 0.5. Something else that you think makes sense)?